File: 音视频/audio/audio-02-audioUnit研究/code/audioStudyDemo/audioStudyDemo/ViewController.m
//
//  ViewController.m
//  audioStudyDemo
//
//  Created by codew on 2018/2/7.
//  Copyright © 2018年 codew. All rights reserved.
//

#import "ViewController.h"
#import "NVAudioAdaptor.h"
@interface ViewController ()


@property (nonatomic, strong) NVAudioAdaptor * adaptor;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    
}

- (void)viewDidAppear:(BOOL)animated
{
    [super viewDidAppear:animated];
    
   [self.adaptor start];
}

- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event
{
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


- (NVAudioAdaptor *)adaptor
{
    
    if (!_adaptor) {
        
        _adaptor = [[NVAudioAdaptor alloc] init];
    }
    return _adaptor;
}

@end

File: iOS/old/重学iOS多线程/code/TestLock/TestLock/ViewController.swift
//
//  ViewController.swift
//  TestLock
//
//  Created by codew on 10/31/18.
//  Copyright © 2018 codew. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        
        
//        NSRecursiveLock'
        
        let ppp = NSRecursiveLock.init();
        
        ppp.lock()
        
        
        ppp.unlock();
        // Do any additional setup after loading the view, typically from a nib.
    }


}

